### PR TITLE
Fix #1397 by calling load_wsgi after starting the reloader

### LINKED ISSUE #1397: Reloader not handling syntax errors any more
When running gunicorn 19.4.x with `--reload` it would trap any "bad" python errors like `SyntaxError` or `IndentationError` and show them on the page.  Very useful.

But in 19.5 and 19.6 it doesn't serve the error message properly, shows an empty page and the logs have this error:

```
Traceback (most recent call last):
  File "/allura-data/virtualenv/local/lib/python2.7/site-packages/gunicorn/workers/sync.py", line 182, in handle_request
    resp.write(item)
  File "/allura-data/virtualenv/local/lib/python2.7/site-packages/gunicorn/http/wsgi.py", line 344, in write
    raise TypeError('%r is not a byte' % arg)
TypeError: ['  File "/allura-data/virtualenv/local/lib/python2.7/site-packages/gunicorn/workers/base.py", line 136, in load_wsgi\n    self.wsgi = self.app.wsgi()\n', '  File "/allura-data/virtualenv/local/lib/python2.7/site-packages/gunicorn/app/base.py", line 67, in wsgi\n    self.callable = self.load()\n', '  File "/allura-data/virtualenv/local/lib/python2.7/site-packages/gunicorn/app/wsgiapp.py", line 63, in load\n    return self.load_pasteapp()\n', '  File "/allura-data/virtualenv/local/lib/python2.7/site-packages/gunicorn/app/wsgiapp.py", line 59, in load_pasteapp\n    return load_pasteapp(self.cfgurl, self.relpath, global_conf=None)\n', '  File "/allura-data/virtualenv/local/lib/python2.7/site-packages/gunicorn/app/pasterapp.py", line 69, in load_pasteapp\n    global_conf=global_conf)\n', '  File "/allura-data/virtualenv/local/lib/python2.7/site-packages/paste/deploy/loadwsgi.py", line 247, in loadapp\n    return loadobj(APP, uri, name=name, **kw)\n', '  File "/allura-data/virtualenv/local/lib/python2.7/site-packages/paste/deploy/loadwsgi.py", line 272, in loadobj\n    return context.create()\n', '  File "/allura-data/virtualenv/local/lib/python2.7/site-packages/paste/deploy/loadwsgi.py", line 710, in create\n    return self.object_type.invoke(self)\n', '  File "/allura-data/virtualenv/local/lib/python2.7/site-packages/paste/deploy/loadwsgi.py", line 146, in invoke\n    return fix_call(context.object, context.global_conf, **context.local_conf)\n', '  File "/allura-data/virtualenv/local/lib/python2.7/site-packages/paste/deploy/util.py", line 56, in fix_call\n    val = callable(*args, **kw)\n', '  File "/allura/Allura/allura/config/middleware.py", line 72, in make_app\n    return _make_core_app(root, global_conf, full_stack, **app_conf)\n', '  File "/allura/Allura/allura/config/middleware.py", line 121, in _make_core_app\n    load_environment(global_conf, app_conf)\n', '  File "/allura-data/virtualenv/local/lib/python2.7/site-packages/tg/configuration.py", line 641, in load_environment\n    self.setup_helpers_and_globals()\n', '  File "/allura-data/virtualenv/local/lib/python2.7/site-packages/tg/configuration.py", line 304, in setup_helpers_and_globals\n    config[\'pylons.app_globals\'] = self.package.lib.app_globals.Globals()\n', '  File "/allura/Allura/allura/lib/app_globals.py", line 295, in __init__\n    tool=_cache_eps(\'allura\', dict_cls=utils.CaseInsensitiveDict),\n', '  File "/allura/Allura/allura/lib/app_globals.py", line 284, in _cache_eps\n    value = ep.load()\n', '  File "/allura-data/virtualenv/local/lib/python2.7/site-packages/pkg_resources/__init__.py", line 2291, in load\n    return self.resolve()\n', '  File "/allura-data/virtualenv/local/lib/python2.7/site-packages/pkg_resources/__init__.py", line 2297, in resolve\n    module = __import__(self.module_name, fromlist=[\'__name__\'], level=0)\n', '  File "/allura/Allura/allura/ext/user_profile/__init__.py", line 18, in <module>\n    from .user_main import UserProfileApp, ProfileSectionBase\n', '  File "/allura/Allura/allura/ext/user_profile/user_main.py", line 35, in <module>\n    from allura.app import Application, SitemapEntry\n', '  File "/allura/Allura/allura/app.py", line 45, in <module>\n    from allura.controllers import BaseController\n', '  File "/allura/Allura/allura/controllers/__init__.py", line 21, in <module>\n    from .discuss import DiscussionController, AppDiscussionController, ThreadController, PostController\n'] is not a byte
```

And in latest master (with `--reload poll` as it seems bare `--reload` no longer works), it throws:

```
[2016-11-18 16:45:32 +0000] [20] [ERROR] Exception in worker process
Traceback (most recent call last):
  File "/allura-data/virtualenv/local/lib/python2.7/site-packages/gunicorn/arbiter.py", line 557, in spawn_worker
    worker.init_process()
  File "/allura-data/virtualenv/local/lib/python2.7/site-packages/gunicorn/workers/base.py", line 115, in init_process
    self.load_wsgi()
  File "/allura-data/virtualenv/local/lib/python2.7/site-packages/gunicorn/workers/base.py", line 157, in load_wsgi
    self.reloader.add_extra_file(exc_val.filename)
AttributeError: 'NoneType' object has no attribute 'add_extra_file'
```

### PR DESCRIPTION
**Pull request by daker**
This resolves the issue introduced in https://github.com/benoitc/gunicorn/commit/64b26ef76648ad2a70f64379f0f5433117450529

### CODE REVIEW

**Comment by brondsem** [comment]
I've tested this and it does fix #1397 for me

File: gunicorn/workers/base.py [hunk 1]
```diff
@@ -156,7 +155,7 @@ def load_wsgi(self):
                 exc_type, exc_val, exc_tb = sys.exc_info()
                 self.reloader.add_extra_file(exc_val.filename)
 
-                tb_string = traceback.format_tb(exc_tb)
+                tb_string = traceback.format_exc(exc_tb)
```
Comment (berkerpeksag) [inline gunicorn/workers/base.py]: Why did you change this? The usage of `format_tb()` was discussed in https://github.com/benoitc/gunicorn/pull/1235#issuecomment-201002065.

Comment (daker) [inline gunicorn/workers/base.py]: With ```traceback.format_tb```, tb_string is an array while [make_fail_app](https://github.com/benoitc/gunicorn/blob/master/gunicorn/util.py#L563) is expecting a string so if you have an error somewhere say in your wsgi file, you get this printed in the console instead showing the real error on the webpage

```
[2016-11-29 19:23:47 +0000] [8872] [ERROR] Error handling request
Traceback (most recent call last):
  File "/home/daker/Projets/guni/env/local/lib/python2.7/site-packages/gunicorn/workers/sync.py", line 182, in handle_request
    resp.write(item)
  File "/home/daker/Projets/guni/env/local/lib/python2.7/site-packages/gunicorn/http/wsgi.py", line 344, in write
    raise TypeError('%r is not a byte' % arg)
TypeError: ['  File "/home/daker/Projets/guni/env/local/lib/python2.7/site-packages/gunicorn/workers/base.py", line 143, in load_wsgi\n    self.wsgi = self.app.wsgi()\n', '  File "/home/daker/Projets/guni/env/local/lib/python2.7/site-packages/gunicorn/app/base.py", line 67, in wsgi\n    self.callable = self.load()\n', '  File "/home/daker/Projets/guni/env/local/lib/python2.7/site-packages/gunicorn/app/wsgiapp.py", line 65, in load\n    return self.load_wsgiapp()\n', '  File "/home/daker/Projets/guni/env/local/lib/python2.7/site-packages/gunicorn/app/wsgiapp.py", line 52, in load_wsgiapp\n    return util.import_app(self.app_uri)\n', '  File "/home/daker/Projets/guni/env/local/lib/python2.7/site-packages/gunicorn/util.py", line 378, in import_app\n    __import__(module)\n'] is not a byte
```

but you only get the real error on the console when gunicorn starts but if you hit the webpage you get nothing(```ERR_CONTENT_LENGTH_MISMATCH```) and the ```...is not a byte``` error is displayed on the console

```
[2016-11-29 19:20:53 +0000] [8867] [INFO] Starting gunicorn 19.6.0
[2016-11-29 19:20:53 +0000] [8867] [INFO] Listening at: http://0.0.0.0:8000 (8867)
[2016-11-29 19:20:53 +0000] [8867] [INFO] Using worker: sync
[2016-11-29 19:20:53 +0000] [8872] [INFO] Booting worker with pid: 8872
[2016-11-29 19:20:53 +0000] [8872] [ERROR] unexpected indent (wsgi.py, line 7)
Traceback (most recent call last):
  File "/home/daker/Projets/guni/env/local/lib/python2.7/site-packages/gunicorn/workers/base.py", line 143, in load_wsgi
    self.wsgi = self.app.wsgi()
  File "/home/daker/Projets/guni/env/local/lib/python2.7/site-packages/gunicorn/app/base.py", line 67, in wsgi
    self.callable = self.load()
  File "/home/daker/Projets/guni/env/local/lib/python2.7/site-packages/gunicorn/app/wsgiapp.py", line 65, in load
    return self.load_wsgiapp()
  File "/home/daker/Projets/guni/env/local/lib/python2.7/site-packages/gunicorn/app/wsgiapp.py", line 52, in load_wsgiapp
    return util.import_app(self.app_uri)
  File "/home/daker/Projets/guni/env/local/lib/python2.7/site-packages/gunicorn/util.py", line 378, in import_app
    __import__(module)
  File "/home/daker/Projets/guni/wsgi.py", line 7
    start_response(status, response_headers)
    ^
IndentationError: unexpected indent
```

Comment (daker) [inline gunicorn/workers/base.py]: it turns out ```format_exc``` takes a limit arg, not a traceback, so this ```tb_string = ''.join(traceback.format_tb(exc_tb))``` would fix the ```...is not a byte``` error but gunicorn will not display the original ```IndentationError``` and display 
```
  File "/home/daker/Projets/guni/env/local/lib/python2.7/site-packages/gunicorn/workers/base.py", line 143, in load_wsgi
    self.wsgi = self.app.wsgi()
  File "/home/daker/Projets/guni/env/local/lib/python2.7/site-packages/gunicorn/app/base.py", line 67, in wsgi
    self.callable = self.load()
  File "/home/daker/Projets/guni/env/local/lib/python2.7/site-packages/gunicorn/app/wsgiapp.py", line 65, in load
    return self.load_wsgiapp()
  File "/home/daker/Projets/guni/env/local/lib/python2.7/site-packages/gunicorn/app/wsgiapp.py", line 52, in load_wsgiapp
    return util.import_app(self.app_uri)
  File "/home/daker/Projets/guni/env/local/lib/python2.7/site-packages/gunicorn/util.py", line 378, in import_app
    __import__(module)
```

**Comment by daker** [comment]
@benoitc can you please quick review this ?

File: gunicorn/workers/base.py (hunk 1 above)
Comment (benoitc) [inline gunicorn/workers/base.py]: i mean let's separate each concerns. The change for load_wsgi is fine for me.

**Comment by berkerpeksag** [comment]
Thanks!

**Pull request merged**